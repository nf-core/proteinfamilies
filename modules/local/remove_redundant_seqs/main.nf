process REMOVE_REDUNDANT_SEQS {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/83/8372f6241b480332d91bc00a88ec8c72c8f7fcc9994177a5dd67a07007cd6e32/data' :
        'community.wave.seqera.io/library/biopython:1.85--6f761292fa9881b4' }"

    input:
    tuple val(meta) , path(clustering)
    tuple val(meta2), path(sequences)

    output:
    tuple val(meta), path("${prefix}.faa"), emit: fasta
    path "versions.yml"                   , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def is_compressed = sequences.getName().endsWith(".gz") ? true : false
    def fasta_name    = sequences.name.replace(".gz", "")
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    if [ "$is_compressed" == "true" ]; then
        gzip -c -d $sequences > $fasta_name
    fi

    remove_redundant_seqs.py \\
        --clustering ${clustering} \\
        --sequences ${fasta_name} \\
        --out_fasta ${prefix}.faa

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version 2>&1 | sed 's/Python //g')
        biopython: \$(python -c "import importlib.metadata; print(importlib.metadata.version('biopython'))")
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.faa

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version 2>&1 | sed 's/Python //g')
        biopython: \$(python -c "import importlib.metadata; print(importlib.metadata.version('biopython'))")
    END_VERSIONS
    """
}
