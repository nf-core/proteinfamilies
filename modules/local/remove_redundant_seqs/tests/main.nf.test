nextflow_process {

    name "Test Process REMOVE_REDUNDANT_SEQS"
    script "../main.nf"
    process "REMOVE_REDUNDANT_SEQS"

    test("proteinfamilies - tsv") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'test_tsv' ], // meta map
                    file(params.pipelines_testdata_base_path + 'proteinfamilies/test_data/modules/clustering.tsv', checkIfExists: true)
                ])
                input[1] = Channel.of([
                    [ id: 'test_fasta' ], // meta map
                    file(params.pipelines_testdata_base_path + 'proteinfamilies/test_data/mgnifams_input_small.fa', checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
            )
        }
    }

    test("proteinfamilies - tsv - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'test_tsv' ], // meta map
                    file(params.pipelines_testdata_base_path + 'proteinfamilies/test_data/modules/clustering.tsv', checkIfExists: true)
                ])
                input[1] = Channel.of([
                    [ id: 'test_fasta' ], // meta map
                    file(params.pipelines_testdata_base_path + 'proteinfamilies/test_data/mgnifams_input_small.fa', checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
