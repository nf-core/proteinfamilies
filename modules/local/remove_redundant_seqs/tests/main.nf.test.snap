{
    "proteinfamilies - tsv": {
        "content": [
            [
                [
                    {
                        "id": "test_tsv"
                    },
                    "test_tsv_null.faa:md5,e06dfa1241372e7fe8aa9f8c20c8f8a9"
                ]
            ],
            [
                {
                    "REMOVE_REDUNDANT_SEQS": {
                        "python": "3.13.1",
                        "biopython": 1.85
                    }
                }
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-29T13:27:03.735181518"
    },
    "proteinfamilies - tsv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_tsv"
                        },
                        "test_tsv_null.faa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,1801b851fa91a19c75b6a4fb3db96f07"
                ],
                "fasta": [
                    [
                        {
                            "id": "test_tsv"
                        },
                        "test_tsv_null.faa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1801b851fa91a19c75b6a4fb3db96f07"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-29T13:27:14.085707025"
    }
}